File: Programming Basics/Nested Loops - Exercises/04.TrainTheTrainers.cs
using System;
using System.Collections.Generic;
using System.Linq;

class Program
{
    static void Main(string[] args)
    {
        List<double> grades = new List<double>();

        int judges = int.Parse(Console.ReadLine());

        string presentation = Console.ReadLine();

        while (presentation != "Finish")
        {
            double currentGrade = 0;

            for (int i = 0; i < judges; i++)
            {
                currentGrade += double.Parse(Console.ReadLine());
            }

            Console.WriteLine($"{presentation} - {currentGrade / judges:f2}.");

            grades.Add(currentGrade / judges);
            presentation = Console.ReadLine();
        }
        Console.WriteLine($"Student's final assessment is {grades.Average():f2}.");
    }
}

File: Programming Basics - Python/Nested Loops - Exercises/04.TrainTheTrainers.py
n = int(input())
presentation = input()

presentations_count = 0
average = 0

while presentation != "Finish":
    average_grade = 0

    for num in range(0, n):
        average_grade += float(input())

    print(f"{presentation} - {average_grade / n:.2f}.")

    average += average_grade / n
    presentations_count += 1

    presentation = input()

print(f"Student's final assessment is {average / presentations_count:.2f}.")